2024-07-11 12:31:52.801550: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:31:52.801763: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:32:01,210 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:32:01,322 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:32:01,322 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:32:01,322 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:32:02,450 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:32:02,450 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:02,467 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44333333333333336}
INFO flwr 2024-07-11 12:32:02,467 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:32:02,793 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:32:02,816 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:32:02,818 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:02,826 | server.py:131 | fit progress: (1, 0.4020841123098278, {'accuracy': 0.8033333333333333}, 0.35896980000000056)
DEBUG flwr 2024-07-11 12:32:02,826 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:32:02,847 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:32:02,847 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:32:02,848 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:32:02,867 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:02,879 | server.py:131 | fit progress: (2, 0.39873172213940594, {'accuracy': 0.81}, 0.41175169999999994)
DEBUG flwr 2024-07-11 12:32:02,879 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:32:02,897 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:32:02,897 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:32:02,918 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:02,930 | server.py:131 | fit progress: (3, 0.39873186407419925, {'accuracy': 0.81}, 0.46275510000000075)
DEBUG flwr 2024-07-11 12:32:02,931 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:32:02,952 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:32:02,952 | server.py:227 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:02,983 | server.py:241 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:02,995 | server.py:131 | fit progress: (4, 0.40188883817740145, {'accuracy': 0.8166666666666667}, 0.5271518000000004)
DEBUG flwr 2024-07-11 12:32:02,995 | server.py:178 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,017 | server.py:192 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,017 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,049 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,061 | server.py:131 | fit progress: (5, 0.401847056498126, {'accuracy': 0.8166666666666667}, 0.5942596000000009)
DEBUG flwr 2024-07-11 12:32:03,061 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,083 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,084 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,117 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,129 | server.py:131 | fit progress: (6, 0.4018435518334067, {'accuracy': 0.8166666666666667}, 0.6619900000000012)
DEBUG flwr 2024-07-11 12:32:03,130 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,156 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,156 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:32:03,190 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,203 | server.py:131 | fit progress: (7, 0.401846117197346, {'accuracy': 0.8166666666666667}, 0.7350767000000005)
DEBUG flwr 2024-07-11 12:32:03,203 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,230 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,231 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,270 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,283 | server.py:131 | fit progress: (8, 0.4028978850368523, {'accuracy': 0.8233333333333334}, 0.8165519999999997)
DEBUG flwr 2024-07-11 12:32:03,284 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,313 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,314 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,356 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,370 | server.py:131 | fit progress: (9, 0.40292491830488814, {'accuracy': 0.8233333333333334}, 0.9028165999999995)
DEBUG flwr 2024-07-11 12:32:03,370 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,401 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:32:03,401 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,434 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:32:03,447 | server.py:131 | fit progress: (10, 0.402923468097017, {'accuracy': 0.8233333333333334}, 0.9797796000000005)
DEBUG flwr 2024-07-11 12:32:03,448 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:32:03,484 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:32:03,484 | server.py:153 | FL finished in 1.0174988000000003
INFO flwr 2024-07-11 12:32:03,485 | app.py:225 | app_fit: losses_distributed [(1, 0.38342320173978806), (2, 0.3830340653657913), (3, 0.4035901298125585), (4, 0.4075779836802256), (5, 0.4075796448049091), (6, 0.4075797200202942), (7, 0.40910031156880516), (8, 0.40871289713042125), (9, 0.408713630267552), (10, 0.40871365921837943)]
INFO flwr 2024-07-11 12:32:03,485 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:32:03,485 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:32:03,485 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4020841123098278), (2, 0.39873172213940594), (3, 0.39873186407419925), (4, 0.40188883817740145), (5, 0.401847056498126), (6, 0.4018435518334067), (7, 0.401846117197346), (8, 0.4028978850368523), (9, 0.40292491830488814), (10, 0.402923468097017)]
INFO flwr 2024-07-11 12:32:03,485 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.44333333333333336), (1, 0.8033333333333333), (2, 0.81), (3, 0.81), (4, 0.8166666666666667), (5, 0.8166666666666667), (6, 0.8166666666666667), (7, 0.8166666666666667), (8, 0.8233333333333334), (9, 0.8233333333333334), (10, 0.8233333333333334)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.10507481,  1.16637907,  0.8058212 ,  0.4901214 ,  0.1795406 ,
         0.05817427, -0.13597928,  0.0290587 ]]), array([0.07634226])]
[array([[-2.10507481,  1.16637907,  0.8058212 ,  0.4901214 ,  0.1795406 ,
         0.05817427, -0.13597928,  0.0290587 ]]), array([0.07634226])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05626414,  1.17235873,  0.76455532,  0.4957691 ,  0.14588041,
         0.10706544, -0.11577732,  0.02289765]]), array([0.02006029])]
[array([[-2.05626414,  1.17235873,  0.76455532,  0.4957691 ,  0.14588041,
         0.10706544, -0.11577732,  0.02289765]]), array([0.02006029])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05663358,  1.17275309,  0.76461475,  0.49619538,  0.14603181,
         0.10687531, -0.11561105,  0.02286523]]), array([0.01998982])]
[array([[-2.05663358,  1.17275309,  0.76461475,  0.49619538,  0.14603181,
         0.10687531, -0.11561105,  0.02286523]]), array([0.01998982])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98469574,  1.04856632,  0.68309214,  0.42619891,  0.21599955,
         0.08831   ,  0.00666703, -0.01506744]]), array([0.05063911])]
[array([[-1.98469574,  1.04856632,  0.68309214,  0.42619891,  0.21599955,
         0.08831   ,  0.00666703, -0.01506744]]), array([0.05063911])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9828873 ,  1.04865965,  0.68276384,  0.42670478,  0.21557169,
         0.08886922,  0.0065359 , -0.0152748 ]]), array([0.04749062])]
[array([[-1.9828873 ,  1.04865965,  0.68276384,  0.42670478,  0.21557169,
         0.08886922,  0.0065359 , -0.0152748 ]]), array([0.04749062])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98264358,  1.04971106,  0.68426221,  0.42731533,  0.21681125,
         0.08969542,  0.0065536 , -0.01477778]]), array([0.04030185])]
[array([[-1.98264358,  1.04971106,  0.68426221,  0.42731533,  0.21681125,
         0.08969542,  0.0065536 , -0.01477778]]), array([0.04030185])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98259276,  1.04971894,  0.68420326,  0.42721192,  0.21687161,
         0.08968375,  0.00652076, -0.01482707]]), array([0.04034554])]
[array([[-1.98259276,  1.04971894,  0.68420326,  0.42721192,  0.21687161,
         0.08968375,  0.00652076, -0.01482707]]), array([0.04034554])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9864817 ,  1.00797139,  0.7005683 ,  0.41007784,  0.19829446,
         0.09803563,  0.04336161,  0.01810855]]), array([0.0350254])]
[array([[-1.9864817 ,  1.00797139,  0.7005683 ,  0.41007784,  0.19829446,
         0.09803563,  0.04336161,  0.01810855]]), array([0.0350254])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98631751,  1.0070409 ,  0.69957877,  0.40935267,  0.19764959,
         0.09742057,  0.04371269,  0.01789793]]), array([0.03973082])]
[array([[-1.98631751,  1.0070409 ,  0.69957877,  0.40935267,  0.19764959,
         0.09742057,  0.04371269,  0.01789793]]), array([0.03973082])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98634847,  1.00707552,  0.69951491,  0.40934983,  0.19765326,
         0.09740944,  0.04369002,  0.01785462]]), array([0.03983893])]
[array([[-1.98634847,  1.00707552,  0.69951491,  0.40934983,  0.19765326,
         0.09740944,  0.04369002,  0.01785462]]), array([0.03983893])]
>>>>>>>>>>>>>run time:
0.5625s
